website/content/en/docs/advanced-topics/heterogeneous.md

```diff
@@ -0,0 +1,400 @@
+[Kubernetes provides a mechanism](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/) called `nodeAffinity` which can be used to limit the possible node targets where a pod can be scheduled. The following example can be used to update a `PodSpec` or `PodTemplateSpec` to prevent the scheduling of pods on nodes of incompatible architectures. Here we compare the `kubernetes.io/arch` key set on the node to ensure that the value matches one of the supported architectures. This assumes that the referenced image points to a manifest list of images for each of the architectures supported. It is important to remember that `nodeAffinity` must be set anywhere a container reference is defined, including `Pod`, `Deployment`, `DaemonSet`, `StatefulSet`, or any other object that defines a `PodSpec` or `PodTemplateSpec`.
+
+```yaml
+affinity:
+  nodeAffinity:
+    requiredDuringSchedulingIgnoredDuringExecution:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: kubernetes.io/arch
+          operator: In
+          values:
+          - amd64
+          - arm64
+          - ppc64le
+          - s390x
+```
+
+The list of architecture values should only include the architectures the operator supports. The full list of possible `GOARCH` values is [available here](https://github.com/golang/go/blob/master/src/go/build/syslist.go). It should be noted that Kubernetes allows a user to specify both `requiredDuringSchedulingIgnoredDuringExecution` and `preferredDuringSchedulingIgnoredDuringExecution`. The syntax for each block is the same. At a minimum, an operator author should set the required terms since they protect against a pod being scheduled on an architecturally incompatible node. If the operator in question has much better performance on a subset of the supported architectures, it may also be prudent to set `preferredDuringSchedulingIgnoredDuringExecution` so that the optimal arches are selected first if they are available.
+
+#### Integrating with Operator Lifecycle Manager
+
+The Operator Lifecycle Manager (OLM) is often used to distribute operators via operator catalogs. The operator deployment object in an OLM integrated operator is defined in the `ClusterServiceVersion` yaml. It is important to remember to set the node affinity block in each of the `spec.template.spec` under `spec.install.deployments`.
+
+## Examples
+This section attempts to show how each of the sample operators could be updated to declare a set of supported architectures. Most of the time, these changes will be in `yaml` files, but different kinds of operators embed `PodSpec` and `PodTemplateSpec` objects in different places.
+
+The most common directory with objects that will need an architecture affinity block is the operator's `config` directory. Additionally, if the operator is configured for OLM, there will likely be a deployment object in the `bundle/manifests/*.clusterserviceversion.yaml`. These locations are common across all operator types.
```

Comment:
That is not the most common. 
The only directory scaffold by default that has manifest is the config/

Ansible/Helm operators will have other directories which are the Ansible playbooks and helm charts. 
So, we might want to clarify that any workload created by the project would need to have the config, for Golang they must check/ensure in the controller/reconciliation code. For Ansible/Helm they will need to ensure that the playbooks and charts are configured accordingly.

Reply:
Is the confusion here is less that `config` is the **most common** and rather that it's the directory that all operator types have **in common** where updates maybe be needed?

Reply:
@jaypoulz the projects use kustomize so the patches ones only add things in the Deployment. However, we have only 1 Deployment with 2 containers/images by default.